index.js

```diff
@@ -44,16 +46,22 @@ module.exports = {
           replacement: "import lodash from 'lodash'\nconst isPlainObject = lodash.isPlainObject"
         }
       ]
-    })
+    });
+
+    reduxTree = esTranspiler(reduxTree, {
```

Comment:
You either need to instruct `transpileTree` to *not* compile modules or you have to merge in the resulting tree after calling _super. As it stands now (in your snippet) you are actually telling it to double transpile modules (which results in nested defines like that).

So the two options are:

```js
let addon = this.addons.find(addon => addon.name === 'ember-cli-babel');
    reduxTree = addon.transpileTree(reduxTree, {
      babel: {
        plugins: ['babel-plugin-transform-object-rest-spread']
      },
      'ember-cli-babel': {
        compileModules: false
      }
    });
```

Or:

```js
    let addon = this.addons.find(addon => addon.name === 'ember-cli-babel');
    reduxTree = addon.transpileTree(reduxTree, {
      babel: {
        plugins: ['babel-plugin-transform-object-rest-spread']
      }
    });

    let superResult = this._super.treeForAddon.call(this, tree);
    return mergeTrees([reduxTree, superResult]);
```